include/envoy/network/connection.h

```diff
@@ -85,6 +86,9 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
     Stats::Gauge& write_current_;
     // Counter* as this is an optional counter. Bind errors will not be tracked if this is nullptr.
     Stats::Counter* bind_errors_;
+    // Counter* as this is an optional counter. Socket options errors will not be tracked if this is
+    // nullptr.
+    Stats::Counter* socket_options_errors_;
```

Comment:
Unless you actually wire this up in production code I would kill it and associated logic. I don't see it wired up anywhere but maybe I missed it.

Reply:
Agree, this is a remnant from the initial straightforward socket mark version. Now the visitor can observe the success/failure of the setsockopt directly, no need for this.